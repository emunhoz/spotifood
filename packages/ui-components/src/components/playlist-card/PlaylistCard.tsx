import React from 'react'
import * as S from './PlaylistCard.style'

interface IPlaylistCard {
  imgUrl: string
  playlistName?: string
  owner?: string
  link?: string
}

export const PlaylistCard = ({
  imgUrl,
  playlistName,
  owner,
  link
}: IPlaylistCard) => (
  <S.PlaylistCard>
    <S.PlaylistCardFigure>
      <S.PlaylistCardImg src={imgUrl} alt='image name' />
    </S.PlaylistCardFigure>
    <S.Footer>
      <S.PlaylistInfo>
        <S.PlaylistInfoTitle>{playlistName}</S.PlaylistInfoTitle>
        <S.PlaylistInfoSubtitle>por {owner}</S.PlaylistInfoSubtitle>
      </S.PlaylistInfo>
      <S.Link href={link} rel='noreferrer' target='_blank'>
        Escutar
      </S.Link>
    </S.Footer>
  </S.PlaylistCard>
)
