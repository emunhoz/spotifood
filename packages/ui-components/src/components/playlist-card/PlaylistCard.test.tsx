import React from 'react'
import { render } from '../../utils/test-utils'
import { PlaylistCard } from './PlaylistCard'

describe('[Component]: PlaylistCard', () => {
  test('render the component', () => {
    const { asFragment } = render(
      <PlaylistCard
        owner='Spotify'
        playlistName='Rain Sounds'
        link='https://open.spotify.com/playlist/37i9dQZF1DX8ymr6UES7vc'
        imgUrl='https://i.scdn.co/image/ab67706f00000003aba1f07094bd3e98cd0122de'
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
