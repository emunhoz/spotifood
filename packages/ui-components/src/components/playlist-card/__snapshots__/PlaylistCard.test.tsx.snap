// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component]: PlaylistCard render the component 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ jXCMpv"
  >
    <figure
      class="sc-gsTCUz bCYJhM"
    >
      <img
        alt="Rain Sounds"
        class="sc-dlfnbm fSNnaC"
        loading="lazy"
        src="https://i.scdn.co/image/ab67706f00000003aba1f07094bd3e98cd0122de"
      />
    </figure>
    <footer
      class="sc-hKgILt jPQFfy"
    >
      <div
        class="sc-eCssSg gotgVw"
      >
        <h6
          class="sc-jSgupP kFMjBk"
        >
          Rain Sounds
        </h6>
        <small
          class="sc-gKsewC JaGrM"
        >
          por Spotify
        </small>
      </div>
      <a
        class="sc-iBPRYJ fJwNpH"
        href="https://open.spotify.com/playlist/37i9dQZF1DX8ymr6UES7vc"
        rel="noreferrer"
        target="_blank"
      >
        Escutar
      </a>
    </footer>
  </div>
</DocumentFragment>
`;
